examples/ethernet/ethernet/main/ethernet_main.c

```diff
@@ -32,92 +32,45 @@
 #include "tcpip_adapter.h"
 #include "nvs_flash.h"
 #include "driver/gpio.h"
-#include "tlk110_phy.h"
 
-static const char *TAG = "eth_demo";
+extern void phy_device_check_phy_init(void);
```

Comment:
I liked (in my version) that ethernet_main did not need to know anything about the underlying phy.  Does it make a difference (other than compile time) whether you do selective compiling or selective linking?